ovgs.proto

```diff
@@ -1,175 +1,241 @@
+message AssignUserRoleRequest {
+  string username = 1;     // username or service account name
+  string user_type = 2;    // user account type
+  string org_id = 3;       // org_id = "org-" + org name
+  string group_id = 4;
+  UserRole user_role = 5;  // role of the user, either ADMIN or REQUESTOR
 }
 
-message assignUserRoleResponse {}
+message AssignUserRoleResponse {
+}
 
-message removeUserRoleRequest {
-    string username = 1;     // username or service account name
-    string user_type = 2;    // "account" for service accounts and "user" for user accounts
-    string org_id = 3;       // org_id = "org-" + org name
-    string group_id = 4;
+message RemoveUserRoleRequest {
+  string username = 1;     // username or service account name
+  AccountType user_type = 2;    // user account type
+  string org_id = 3;       // org_id = "org-" + org name
+  string group_id = 4;
 }
 
-message removeUserRoleResponse {}
+message RemoveUserRoleResponse {
+}
 
-message viewUserRoleRequest {
-    string username = 1;     // username or service account name
-    string user_type = 2;    // "account" for service accounts and "user" for user accounts
-    string org_id = 3;       // org_id = "org-" + org name
+message GetUserRoleRequest {
+  string username = 1;     // username or service account name
+  string user_type = 2;    // user role type
+  string org_id = 3;       // org_id = "org-" + org name
 }
 
-message viewUserRoleResponse {
-    map<string, userRole> group_to_role = 1;  // mapping from a group to user's role in that group
+message GetUserRoleResponse {
+  // mapping from a group to user's role in that group
+  map<string, UserRole> group_to_role = 1;
 }
 
-message setDomainCertRequest {
-    string group_id = 1;
-    bytes certificate_der = 2;
-    bool revocation_checks = 3;
-    google.protobuf.Timestamp expiry_time = 4;
+message CreateDomainCertRequest {
+  string group_id = 1;
+  bytes certificate_der = 2;
+  bool revocation_checks = 3;
+  google.protobuf.Timestamp expiry_time = 4;
 }
 
-message setDomainCertResponse {
-    string cert_id = 1;
+message CreateDomainCertResponse {
+  string cert_id = 1;
 }
 
-message readDomainCertRequest {
-    string cert_id = 1;
+message GetDomainCertRequest {
+  string cert_id = 1;
 }
 
-message readDomainCertResponse {
-    string group_id = 1;
-    bytes certificate_der = 2;
-    bool revocation_checks = 3;
-    google.protobuf.Timestamp expiry_time = 4;
+message GetDomainCertResponse {
+  string cert_id = 1;
+  string group_id = 2;
+  bytes certificate_der = 3;
+  bool revocation_checks = 4;
+  google.protobuf.Timestamp expiry_time = 5;
 }
 
-message deleteDomainCertRequest {
-    string cert_id = 1;
+message DeleteDomainCertRequest {
+  string cert_id = 1;
 }
 
-message deleteDomainCertResponse {}
+message DeleteDomainCertResponse {
+}
 
-message assignSerialRequest {
-    string switch_serial = 1;
-    string group_id = 2;
+message CreateSerialRequest {
```

Comment:
Can we stick to AssignSerial?  Same for UnassignSerial.
The rationale is the following - All serials owned by customer X will show up in the root group, which is pre-created and has a group_id that looks like = org-X. If you invoke viewGroup on the root group, it will show all serials owned by you and you can subsequently move/assign them from the root group to one of its children. 

Create serial sounds off, since what you're doing is assigning serials (that already show up in the root group)

Reply:
sure. AddSerial is good.